virttest/virt_vm.py

```diff
@@ -1206,33 +1208,28 @@ def wait_for_serial_login(self, timeout=LOGIN_WAIT_TIMEOUT,
         :params virtio: is a virtio console.
         :return: A ShellSession object.
         """
-        error_messages = []
         logging.debug("Attempting to log into '%s' via serial console "
                       "(timeout %ds)", self.name, timeout)
-        end_time = time.time() + timeout
+        end_time, exception = time.time() + timeout, None
         while time.time() < end_time:
             try:
```

Comment:
If N time sucess, session will return and exception the last line will not be execute.